docs/design.html

```diff
@@ -290,24 +290,56 @@ <h3 class="anchor-heading"><a id="semantics" class="anchor-link"></a><a href="#s
     <p>
-    So effectively Kafka supports exactly-once delivery in <a href="https://kafka.apache.org/documentation/streams">Kafka Streams</a>, and the transactional producer/consumer can be used generally to provide
+    As a result, Kafka supports exactly-once delivery in <a href="https://kafka.apache.org/documentation/streams">Kafka Streams</a>, and the transactional producer/consumer can be used generally to provide
     exactly-once delivery when transferring and processing data between Kafka topics. Exactly-once delivery for other destination systems generally requires cooperation with such systems, but Kafka provides the
     offset which makes implementing this feasible (see also <a href="https://kafka.apache.org/documentation/#connect">Kafka Connect</a>). Otherwise, Kafka guarantees at-least-once delivery by default, and allows
     the user to implement at-most-once delivery by disabling retries on the producer and committing offsets in the consumer prior to processing a batch of messages.
 
-    <h3 class="anchor-heading"><a id="replication" class="anchor-link"></a><a href="#replication">4.7 Replication</a></h3>
+    <h3 class="anchor-heading"><a id="usingtransactions" class="anchor-link"></a><a href="#usingtransactions">4.7 Using Transactions</a></h3>
+    <p>
+    As mentioned above, the simplest way to get exactly-once semantics from Kafka is to use <a href="https://kafka.apache.org/documentation/streams">Kafka Streams</a>. However, it is also possible to achieve
+    the same transactional guarantees using the Kafka producer and consumer directly by using them in the same way as Kafka Streams does.
+    <p>
+    Kafka transactions are a bit different than transactions in other messaging systems. In Kafka, the consumer and producer are separate and it is only the producer which is transactional. It is however able to
+    make transactional updates to the consumer's position (confusingly called the "committed offset"), and it is this which gives the overall exactly-once behavior.
+    <p>
+    There are three key aspects to exactly-once processing using the producer and consumer, which match how Kafka Streams works.
+    <ol>
+        <li>The consumer uses partition assignment to ensure that it is the only consumer in the consumer group currently processing each partition.</li>
+        <li>The consumer uses read-committed isolation level to ensure that it does not consume records produced by transactions which aborted.</li>
+        <li>The producer uses transactions so that all of the records it produces, and any offsets it updates on behalf of the consumer, are performed atomically.</li>
+    </ol>
+    <p>
+    The consumer configuration must include <code>isolation.level=read_committed</code> and <code>enable.auto.commit=false</code>. The producer configuration must set <code>transactional.id</code>
```

Comment:
I don't think the consumer necessarily needs to be read_committed unless it is also reading from a topic written via transactions.

Reply:
Yes, that's true but it's a weird transactional application that wants to process aborted data.

Reply:
It could also be a topic not written with transactions and in that case, read committed and read uncommitted have the same behavior. 

Mostly just calling out it isn't a requirement, though probably a good practice.

Reply:
Seems, I did not split hairs above :)